matrix_client/client.py

```diff
@@ -161,6 +161,27 @@ def login_with_password_no_sync(self, username, password):
         self.api.token = self.token
         return self.token
 
+    def login_with_jwt_no_sync(self, jwt):
```

Comment:
I think this should be rolled into the `login_with_jwt` method. Just add a bool
`sync` kwarg that defaults to `True`. I know this doesn't follow what's been
established with `login_with_password` method, but if we add a lot of new login
methods in the future, it will be cleaner to have just a single method for each
kind of login.